input/docs/handbook/data-binding/xamarin-forms.md

```diff
@@ -1,63 +1,50 @@
-In the app.xaml file register your viewmodels (If you want the viewmodel based navigation)
-```    
-this.Router = new RoutingState();
-Locator.CurrentMutable.RegisterConstant(this, typeof(IScreen));
-Locator.CurrentMutable.Register(() => new LoginPage(), typeof(IViewFor<LoginViewModel>));
-Locator.CurrentMutable.Register(() => new RegisterPage(), typeof(IViewFor<RegisterViewModel>));
-```
-Include the splat package for Locator (Service locator) class (https://github.com/paulcbetts/splat). To start the app and navigate to the default page login page in our case use the below code
-```
-this.Router.Navigate.Execute(new HomeViewModel());
-MainPage = new RoutedViewHost();
-```
-You can implement logic here checking if the user has a valid token and may be navigate to the home page bypassing the login page. It could be a check with in the akavache storage for the presense of a token.
+# Xamarin Forms
 
-Below is the sample loginviewmodel
+For Xamarin.Forms applications, you need to implement `IViewFor<T>` by hands and ensure that ViewModel is a `BindableProperty`. You can use base classes containing default `IViewFor` implementation from `ReactiveUI.XamForms` package, such as `ReactiveContentPage<TViewModel>`, `ReactiveTextCell<TViewModel>`, etc. Also, always dispose bindings via [WhenActivated](../when-activated), or else the bindings leak memory.
 
-```
-public LoginViewModel(IScreen screen = null)
+The goal in the example below is to two-way bind `TheText` property of `TheViewModel` to the Entry and one-way bind `TheText` property to the Label, so the Label updates when the user types text into the Entry.
+ 
+```csharp
+public class TheViewModel : ReactiveObject
```

Comment:
I think the original example shows a little bit more of the ReactiveUI API surface.  I can provide a sample XAML page and code behind file for bindings if needs be.

Reply:
Sure, if it demonstrates the usage of `IViewFor`, `WhenActivated` and `Binding` features for Xamarin.Forms. Considering Routing, might be worth opening a separate PR and start working on a comprehensive guide there.

Reply:
I am going to look at the routing documentation this weekend.  Specifically, it shows more binding options, `OneWayBind`, `TwoWayBind` and `BindCommand`.  I'd rather show more in terms of the Binding Extensions than less.

I'll draft something tonight and just push it to this PR, sound good?  It will add XAML and a XAML.cs view of what the page bindings look like which I feel is what is missing from the original example.  It just doesn't show how to setup bindings on the page.

Reply:
Yeah, sounds great. Thank you!